### PR TITLE
Raise an error if repo used in image meta is undefined

### PR DESCRIPTION
**Pull request by vfreex**
Some CVP content set errors are caused by misspelled repo names in image metas. However we didn't really validate them.

### CODE REVIEW

File: doozerlib/repos.py [hunk 1]
```diff
@@ -271,7 +271,10 @@ def content_sets(self, shipping_repos=[]):
         """Generates a valid content_sets.yml file based on the currently
         configured and enabled repos in the collection. Using the correct
         name for each arch."""
-
+        # check for missing repos
+        missing_repos = set(shipping_repos) - self._repos.keys()
+        if missing_repos:
+            raise ValueError(f"Image meta references undefined repo(s): {missing_repos}")
```
Comment (joepvd) [inline doozerlib/repos.py:277]: `s/Image meta/enabled_repos/`

**Review by joepvd — COMMENTED** [review]
:thinking: Maybe the validator is a more appropriate place for this type of internal consistency check?

File: doozerlib/repos.py (hunk 1 above)
Comment (vfreex) [reply doozerlib/repos.py]: updated. next push

**Review by joepvd — APPROVED** [review]
/lgtm

:shipit:

**Pull request merged**